### PR TITLE
feat: vector conditioning in ABUPT

### PR DESCRIPTION
**Pull request by HennerM**
ABUPT conditioning is just modulation from a linear projection of the global input features. We have a `ScalarConditioner` class that is using fourier-features, a MLP per feature and a shared MLP to embed global features instead, which is strictly more expressive than the simple linear layer. 

However, ScalarsConditioner assumes each global feature is a scaler. To better match semantics of DataSpecs, it makes mroe sense to extend this to VectorsConditioner, which operates on a dictionary of vectors. This is more general and can include vectors of len=1. The other difference is ScalarsConditioner was using the ContinuousSincCosEmbed with the frequency bands optimized for integer values and depending on min-max normalization with a scale in the right range, which is not always the best default. Instead, VectorsConditioner assuems the input is in [-1,1] range and uses NeRF like frequency bands instead.

### CODE REVIEW

File: src/noether/core/configs/hyperparameters.py [hunk 1]
```diff
@@ -43,5 +43,5 @@ def log(stage_hyperparameters: BaseModel) -> None:
         Returns:
             None
         """
-        yaml_str = yaml.dump(stage_hyperparameters.model_dump()).rstrip("\n")
+        yaml_str = yaml.dump(stage_hyperparameters.model_dump(exclude_computed_fields=True)).rstrip("\n")
```
Comment (MauritsBleeker) [inline src/noether/core/configs/hyperparameters.py:83]: I thought we already had `exclude_computed_fields=True`, or is this somewhere else?

File: src/noether/core/schemas/models/ab_upt.py [hunk 2]
```diff
@@ -107,6 +110,14 @@ class AnchorBranchedUPTConfig(ModelBaseConfig, InjectSharedFieldFromParentMixin)
...
     """Drop path rate for stochastic depth. Defaults to 0.0 (no drop path)."""
 
+    geometry_conditioning_dims: FieldDimSpec | None = None
+    """Per-named-field conditioning spec for geometry transformer blocks. When
+    left unset, defaults to ``data_specs.conditioning_dims`` so the geometry
+    branch sees the same conditioning as the rest of the model. An explicit
+    empty :class:`FieldDimSpec` (``total_dim == 0``) opts out — useful for
+    diffusion, where timestep modulation should touch physics + per-domain
+    decoders but not the geometry branch (geometry is invariant to noise level)."""
+
```
Comment (MauritsBleeker) [inline src/noether/core/schemas/models/ab_upt.py:120]: Do we only want conditioning regarding the geometry, or all global conditioning (which does not depend on the geometry)?

File: src/noether/modeling/models/ab_upt.py [hunk 3]
```diff
@@ -28,6 +31,32 @@
...
 
 
+class ReadoutLayer(nn.Module):
+    """
+    The final readout layer for AB-UPT, which applies an AdaLN-style modulation followed by a linear projection to the desired output dimension.
+    """
+
+    def __init__(self, decoder_config: LinearProjectionConfig, hidden_dim: int, condition_dim: int | None = None):
+        super().__init__()
+        self.norm_final = nn.LayerNorm(hidden_dim, elementwise_affine=condition_dim is None, eps=1e-6)
```
Comment (MauritsBleeker) [inline src/noether/modeling/models/ab_upt.py:41]: nit; just call it `self.norm`?

**Review by MauritsBleeker — APPROVED** [review]

File: src/noether/core/schemas/models/ab_upt.py (hunk 2 above)
Comment (HennerM) [reply src/noether/core/schemas/models/ab_upt.py:120]: Sorry I don't understand the question. Having dedicated geometry conditioning is useful, because that allows us to reuse the geometry encoding for samples that have the same geometry but e.g. different inflow-conditions. So basically geometry condition can be a subset of the global conditioning. Does this make sense?
